### ISSUE TITLE
Better API documentation for `TimeoutMixin`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4684]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#4684|
|Type|enhancement|
|Created|2010-10-12 15:13:48Z|
|Branch|https://github.com/twisted/twisted/tree/timeoutmixin-doc-4684|

The API documentation for `TimeoutMixin` isn't very helpful for someone who is not familiar with the code.

In particular, `resetTimeout`'s purpose and its behavior for edge cases are unclear.

<details><summary>Searchable metadata</summary>

```
trac-id__4684 4684
type__enhancement enhancement
reporter__jml jml
priority__normal normal
milestone__ 
branch__branches_timeoutmixin_doc_4684 branches/timeoutmixin-doc-4684
branch_author__jml jml
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1286896428000000 1286896428000000
changetime__1286965323000000 1286965323000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

(In [30136]) Branching to 'timeoutmixin-doc-4684'

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml commented|
|-|-|

(In [30137]) Improvements for `TimeoutMixin` documentation

refs [#4684](https://github.com/twisted/twisted/issues/4684)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml commented|
|-|-|

(In [30138]) NEWS turd. refs [#4684](https://github.com/twisted/twisted/issues/4684).

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @jml|
|-|-|

  * The news fragment doesn't follow the style recommended on http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles
  * Please update the copyright date in policies.py

That's all, please merge when fixed.  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml commented|
|-|-|

(In [30139]) Conform to style guide; update copyright. refs [#4684](https://github.com/twisted/twisted/issues/4684)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set status to `closed`|
|-|-|

(In [30140]) Merge timeoutmixin-doc-4684

 * Author: jml
 * Reviewer: exarkun
 * Fixes [#4684](https://github.com/twisted/twisted/issues/4684)

Improve the API documentation for `TimeoutMixin`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
